api/utils/gcp/endpoint.go

```diff
@@ -21,9 +21,17 @@ import (
 const (
 	// gcpEndpointSuffix is hostname suffix used to determine if hostname is a GCP endpoint
 	gcpEndpointSuffix = ".googleapis.com"
+
+	// SpannerEndpoint is the endpoint that all spanner RPCs should go to.
+	SpannerEndpoint = "spanner" + gcpEndpointSuffix + ":443"
 )
 
 // IsGCPEndpoint returns true if hostname is a GCP endpoint
 func IsGCPEndpoint(hostname string) bool {
 	return strings.HasSuffix(hostname, gcpEndpointSuffix)
 }
+
+// IsSpannerEndpoint returns true if the input URI is the Spanner API endpoint.
+func IsSpannerEndpoint(uri string) bool {
+	return strings.Contains(uri, SpannerEndpoint)
```

Comment:
I'm very tempted to just make it only accept `spanner.googleapis.com:443`.  I really haven't seen anything else that should be used. If anyone ever asks for something custom (no idea what usecase that would be), then we could relax it more.

So how about I change it to just match the exact string instead of `strings.Contains`?

Reply:
Exact match sounds good. As you say, we can always relax it as needed.

Reply:
done in6520f1281f5bad5a5995b0d258b7ffe806575071